src/network/networkd-dhcp4.c

```diff
@@ -147,9 +149,17 @@ static int link_set_dhcp_routes(Link *link) {
 
         n = sd_dhcp_lease_get_routes(link->dhcp_lease, &static_routes);
         if (n == -ENODATA)
-                return 0;
-        if (n < 0)
-                return log_link_warning_errno(link, n, "DHCP error: could not get routes: %m");
+                log_link_info(link, "DHCP: No routes received from DHCP server: %m");
+        else if (n < 0)
+                log_link_warning_errno(link, n, "DHCP error: could not get routes: %m");
+
+        if (n < 0 && r < 0) {
+
+                log_link_warning(link, "DHCP error: No routes received from DHCP server: %m");
+
+                link->dhcp4_configured = true;
+                link_check_ready(link);
+        }
```

Comment:
so, even though it's not strictly necessary either you should do "return 0" here i figure, or just move the whole block to the end of the function... otherwise it's weird, if you still go through the loop even though you already decided its unnecessary...

or am i missing something here?

Reply:
Ok done